Documentation/ApiOverview/TSFE/Index.rst

```diff
@@ -0,0 +1,362 @@
+For more details it is best to look in the source code.
+
+There are at least 3 contexts in which the term TSFE is used:
+
+1.  It was and is available as global array :php:`$GLOBALS['TSFE']` in PHP.
+2.  TypoScript function :ref:`TSFE <t3tsref:data-type-gettext-tsfe>` which can
+    be used to access public properties in TSFE.
+3.  (deprecated since v9.5) Page TSconfig :typoscript:`TSFE.constants`.
+
+Focusing on the PHP part (as the TypoScript part is covered in the TypoScript
+Reference), this page gives an overview, what can still be used, what is deprecated,
+removed or discouraged and in which way the TSFE class itself may be
+interesting to developers.
+
+Acccessing TSFE
+===============
+
+.. note::
+
+    Some of the former public properties and methods have been changed to
+    protected or marked as internal. You may still find code examples online
+    where :php:`$GLOBALS['TSFE']` is used. In several (but not all) cases, there
+    are alternatives available.
+
+From the source:
+
+    When calling a Frontend page, an instance of this object is available
+    as :php:`$GLOBALS['TSFE']`, even though the core development strives to get
+    rid of this in the future.
+
+:php:`$GLOBALS['TSFE']` is not available in all contexts. In particular, it is
+only available in frontend contexts, not in the backend or cli.
+
+Initializing :php:`$GLOBALS['TSFE']` in the Backend is sometimes done in code
+examples found online. TSFE is not initialized in the backend context by the
+core (and there is usually no need to do this).
+
+From the PHP documentation:
+
+    As of PHP 8.1.0, $GLOBALS is now a read-only copy of the global symbol table.
+    That is, global variables cannot be modified via its copy.
+
+https://www.php.net/manual/en/reserved.variables.globals.php
+
+Howtos
+======
+
+.. add howtos for common use cases, remove the not recommended, deprecated
+.. way if there is a better way
+
+Following are some examples which use TSFE and alternatives to using TSFE,
+where available:
+
+-------------
+
+Access the :php:`ContentObjectRenderer` (often referred to as "cObj"):
+
+.. code-block:: php
+
+    $GLOBALS['TSFE']->cObj
+
+Can be done as follows in Extbase controllers:
+
+.. code-block:: php
+
+    $contentObj = $this->configurationManager->getContentObject();
+
+-------------
+
+Access the current page id:
+
+.. code-block:: php
+
+    $GLOBALS['TSFE']->id
+
+Can be done using the ContentObjectRenderer (see previous example):
+
+.. code-block:: php
+
+    $pageId = $contentObj->data['pid'];
+
+-------------
+
+Access language settings is now done using the
+:ref:`language aspect <context_api_aspects_language>`.
+
+Get the request language of the current page as integer:
+
+.. code-block:: php
+
+    $context->getPropertyFromAspect('language', 'id');
+
+--------------
+
+Accessing information about Frontend and Backend users.
+
+.. code-block:: php
+
+    $feUser = $GLOBALS['TSFE']->fe_user;
```

Comment:
Request attribute is another possibility (as this is mostly available in Extbase controllers): https://docs.typo3.org/m/typo3/reference-coreapi/11.5/en-us/ApiOverview/RequestLifeCycle/RequestAttributes/FrontendUser.html

Reply:
done. Thanks again for the link.